### PR TITLE
remove old implementation of command_argument_count

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/4593

### CODE REVIEW

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -1199,11 +1199,17 @@ namespace CommandArgumentCount {
             x.base.base.loc, diagnostics);
     }
 
-    static inline ASR::asr_t* create_CommandArgumentCount(Allocator& al, const Location& loc, Vec<ASR::expr_t*>& /*args*/, diag::Diagnostics& diag) {
+    static ASR::expr_t *eval_CommandArgumentCount(Allocator &al, const Location &loc,
+            ASR::ttype_t* /*t1*/, Vec<ASR::expr_t*> &/*args*/, diag::Diagnostics& /*diag*/) {
+                ASRBuilder b(al, loc);
+        return b.i32(0);
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:1205]: Although this is wrong I think but I'm not sure about how to handle this...
https://github.com/lfortran/lfortran/pull/4863#issuecomment-2366769387

**Comment by HarshitaKalani** [comment]
```fortran
program test_get_command_argument
implicit none
    integer :: cnt 
    cnt = command_argument_count()
    print *, cnt

end program
```
The above program throws error:
```console
$ lfortran a.f90
Traceback (most recent call last):
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in _start
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 2919, in main
    return main_app(argc, argv);
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 2862, in main_app(int, char**)
    err = compile_src_to_object_file(arg_file, tmp_o, false,
  File "/home/harshita/Desktop/lfortran/src/bin/lfortran.cpp", line 1293, in (anonymous namespace)::compile_src_to_object_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, LCompilers::CompilerOptions&, LCompilers::PassManager&)
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/home/harshita/Desktop/lfortran/src/lfortran/fortran_evaluator.cpp", line 435, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/home/harshita/Desktop/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 10155, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    if (llvm::verifyModule(*v.module, &err)) {
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::verifyModule(llvm::Module const&, llvm::raw_ostream*, bool*)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::InstVisitor<(anonymous namespace)::Verifier, void>::visit(llvm::Instruction&)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in (anonymous namespace)::Verifier::visitStoreInst(llvm::StoreInst&)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::VerifierSupport::Write(llvm::Value const&)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::Value::print(llvm::raw_ostream&, llvm::ModuleSlotTracker&, bool) const
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in (anonymous namespace)::AssemblyWriter::printInstruction(llvm::Instruction const&)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in WriteAsOperandInternal(llvm::raw_ostream&, llvm::Value const*, (anonymous namespace)::TypePrinting*, llvm::SlotTracker*, llvm::Module const*)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::SlotTracker::getLocalSlot(llvm::Value const*)
  Binary file "/home/harshita/Desktop/lfortran/src/bin/lfortran", in llvm::SlotTracker::processModule()
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", in <invalid>
Segfault: Signal SIGSEGV (segmentation fault) received
```

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 2]
```diff
@@ -2014,6 +2014,9 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 }
                 break ;
             }
+            case ASRUtils::IntrinsicElementalFunctions::CommandArgumentCount: {
+                break;
+            }
```
Comment (Pranavchiku) [inline src/libasr/codegen/asr_to_llvm.cpp:2019]: Now we will never have this case.

**Review by HarshitaKalani — COMMENTED** [review]
Thanks @Pranavchiku . This looks great!

**Pull request merged**